File: agents/clean-code-analyzer/scripts/tests/e2e/fixtures/dirty_go/solid_violations.go
// KNOWN VIOLATIONS: SOLID-S=1, Naming=2, ErrorHandling=2

package main

// SRP violation: God struct handles multiple concerns
type GodService struct {
    db interface{}
}

func (s *GodService) CreateUser(n string) {}      // Does user AND email AND reporting
func (s *GodService) SendEmail(e string) {}
func (s *GodService) GenerateReport(d interface{}) {}
func (s *GodService) BackupDB() {}

func process(d interface{}) interface{} {  // Naming: unclear
    x := 86400  // Naming: magic number
    _ = x
    err := doRiskyThing()
    if err != nil {
        _ = err  // ErrorHandling: error ignored
    }
    err2 := doAnotherThing()
    _ = err2    // ErrorHandling: error ignored
    return nil
}

func doRiskyThing() error { return nil }
func doAnotherThing() error { return nil }

File: agents/clean-code-analyzer/scripts/tests/e2e/fixtures/dirty_typescript/solid_violations.ts
// KNOWN VIOLATIONS: SOLID-S=2, SOLID-D=1, Naming=2, ErrorHandling=1

class GodService {
  // SRP violation: handles users, emails, AND payments
  createUser(name: string): void { }
  sendWelcomeEmail(email: string): void { }
  processPayment(amount: number): void { }
  generateInvoice(orderId: string): void { }
  backupDatabase(): void { }
}

class OrderProcessor {
  private db = new MySQLDatabase();  // DIP violation: concrete dependency

  process(order: any): any {  // Naming: 'any' type, unclear method
    try {
      return this.db.save(order);
    } catch (e) {
      // ErrorHandling: swallowed silently
    }
  }
}

const x = 86400;  // Naming: magic number
